avm/ptn/lz/sub-vending/modules/subscriptionAlias.bicep

```diff
@@ -26,21 +26,42 @@ param subscriptionTenantId string = ''
 @description('''The Azure Active Directory principals object ID (GUID) to whom should be the Subscription Owner. Leave blank unless following this scenario only [Programmatically create MCA subscriptions across Azure Active Directory tenants](https://learn.microsoft.com/azure/cost-management-billing/manage/programmatically-create-subscription-microsoft-customer-agreement-across-tenants).''')
 param subscriptionOwnerId string = ''
 
+@description('Whether to move the subscription to the specified management group supplied in the pararmeter subscriptionManagementGroupId.')
+param subscriptionManagementGroupAssociationEnabled bool = true
+
+@description('The destination management group ID for the new subscription. Note: Do not supply the display name. The management group ID forms part of the Azure resource ID. e.g., `/providers/Microsoft.Management/managementGroups/{managementGroupId}`.')
+param subscriptionManagementGroupId string = ''
+
+var formattedAdditionalProperties = union(
+  subscriptionManagementGroupAssociationEnabled && !empty(subscriptionManagementGroupId)
+    ? {
+        managementGroupId: tenantResourceId('Microsoft.Management/managementGroups', subscriptionManagementGroupId)
+      }
+    : {},
+  (!empty(subscriptionTenantId) && !empty(subscriptionOwnerId))
+    ? {
+        subscriptionTenantId: subscriptionTenantId
+        subscriptionOwnerId: subscriptionOwnerId
+      }
+    : {}
+)
+
 resource subscriptionAlias 'Microsoft.Subscription/aliases@2021-10-01' = {
   scope: tenant()
   name: subscriptionAliasName
   properties: {
     workload: subscriptionWorkload
     displayName: subscriptionDisplayName
     billingScope: subscriptionBillingScope
-    additionalProperties: (!empty(subscriptionTenantId) && !empty(subscriptionOwnerId)) ? {
-      subscriptionTenantId: subscriptionTenantId
-      subscriptionOwnerId: subscriptionOwnerId
-    } : {}
+    additionalProperties: formattedAdditionalProperties
```

Comment:
From what I understand of this RP, moving this to here makes it a WORM style operation and therefore you cannot update the Management Group Association after the initial creation.

Hence why we do it as a separate resource/operation today to ensure you can move a subscription later down the line, if you so wish.

Reply:
Hi @jtracey93 , sorry for the late response! Due to personal reasons, I couldn't get back to you earlier. All is well now.

Thank you for the explanation and logic behind doing it as a separate resource/operation. When I opened this PR, I thought of it more as an improvement to the code. I did not think it was necessary to call a separate resource. My change does the "same," and if you want to move the subscription later down the line, you can stil call the main.bicep with a target management group.